contracts/pods/VetoController.sol

```diff
@@ -0,0 +1,143 @@
+    }
+
+    event VetoTimelock(uint256 indexed podId, address indexed timelock);
+    event GrantVetoPermission(
+        uint256 indexed podId,
+        address indexed timelock,
+        bytes32 tribeRole
+    );
+    event RevokeVetoPermission(
+        uint256 indexed podId,
+        address indexed timelock,
+        bytes32 tribeRole
+    );
+
+    /// @notice Mapping between a pod timelock and the TribeRoles that are allowed to veto transactions on it
+    mapping(address => EnumerableSet.Bytes32Set) private podVetoRoles;
+
+    /// @notice Get all the TribeRoles that have veto power over a pod
+    function getPodVetoRoles(uint256 _podId)
+        external
+        view
+        returns (bytes32[] memory)
+    {
+        address timelock = podFactory.getPodTimelock(_podId);
+
+        if (podVetoRoles[timelock].length() == 0) {
+            bytes32[] memory empty = new bytes32[](0);
+            return empty;
+        }
+        return podVetoRoles[timelock].values();
+    }
+
+    /// @notice Grant veto permission over a particular pod to a TribeRole
+    /// @dev Permissioned to the GOVERNOR, POD_VETO_ADMIN and GUARDIAN. Used to allow particular TribeRoles
+    ///      to be granted veto ability over pods
+    function grantVetoPermission(uint256 _podId, bytes32 role)
+        public
+        hasAnyOfThreeRoles(
+            TribeRoles.GOVERNOR,
+            TribeRoles.POD_VETO_ADMIN,
+            TribeRoles.GUARDIAN
+        )
+    {
+        address timelock = podFactory.getPodTimelock(_podId);
+        podVetoRoles[timelock].add(role);
+        emit GrantVetoPermission(_podId, timelock, role);
+    }
+
+    /// @notice Revoke a particular TribeRole's veto permission over a pod
+    function revokeVetoPermission(uint256 _podId, bytes32 role)
+        public
+        hasAnyOfThreeRoles(
+            TribeRoles.GOVERNOR,
+            TribeRoles.POD_VETO_ADMIN,
+            TribeRoles.GUARDIAN
+        )
+    {
+        address timelock = podFactory.getPodTimelock(_podId);
+        podVetoRoles[timelock].remove(role);
+        emit RevokeVetoPermission(_podId, timelock, role);
+    }
+
+    /// @notice Batch grant veto permission
+    function batchGrantVetoPermission(
+        uint256[] memory _podIds,
+        bytes32[] memory roles
+    ) external {
+        require(_podIds.length == roles.length, "Mismatched input lengths");
+        for (uint256 i = 0; i < roles.length; ) {
+            grantVetoPermission(_podIds[i], roles[i]);
+
+            // i is constrained by being < roles.length
+            unchecked {
+                i += 1;
+            }
+        }
+    }
+
+    /// @notice Batch revoke veto permission
+    function batchRevokeVetoPermission(
+        uint256[] memory _podIds,
+        bytes32[] memory roles
+    ) external {
+        require(_podIds.length == roles.length, "Mismatched input lengths");
+        for (uint256 i = 0; i < roles.length; ) {
+            revokeVetoPermission(_podIds[i], roles[i]);
```

Comment:
I would recommend the same pattern here of removing the check that does multiple calls to core to check permissions, put logic in an internal helper function and call that function instead and permission this function explicitly.